dali/util/odirect_file.cc

```diff
@@ -101,12 +101,18 @@ ptrdiff_t ODirectFileStream::TellRead() const {
 }
 
 size_t ODirectFileStream::ReadAt(void * buffer, size_t n_bytes, off_t offset) {
-  size_t n_read = pread(fd_, buffer, n_bytes, offset);
+  auto n_read = pread(fd_, buffer, n_bytes, offset);
+  static_assert(std::is_same_v<decltype(n_read), ssize_t>);
```

Comment:
Do we expect to have a different return type anywhere?

Reply:
Not really. In the wake of the initial problem, I wanted to have explicit compile-time check. If you think it is too explicit, I can go with just `size->ssize_t` change.